server/src/main/java/org/opensearch/common/util/FeatureFlags.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.util;
+
+/**
+ * Utility class to manage feature flags. Feature flags are system properties that must be set on the JVM.
+ * These are used to gate the visibility/availability of incomplete features. Fore more information, see
+ * https://featureflags.io/feature-flag-introduction/
+ */
+public class FeatureFlags {
+
+    /**
+     * Gates the visibility of the index setting that enables segment replication.
+     * Once the feature is ready for production release, this feature flag can be removed.
+     */
+    public static final String SEGREP_FEATURE_FLAG = "opensearch.segment_replication_feature_flag_enabled";
```

Comment:
More annoying nitpicks, but I'd consider scoping this differently such that all feature flags would have a common namespace, such as:

```
opensearch.experimental.feature.segment_replication = on;
```

on/off reads more naturally to me but true/false is fine too.

Reply:
I prefer the use of true/false as the value, so I'll change the feature flag to:

```
opensearch.experimental.feature.replication_type.enabled
```